### PR TITLE
Instantiate everything also for complex-valued vectors.

### PR DESCRIPTION
**Pull request by bangerth**
After all of the previous patches (see the links in #2033), this finally makes it
possible to instantiate everything also for complex-valued vectors. Do so.

### CODE REVIEW

File: cmake/config/template-arguments.in [hunk 1]
```diff
@@ -32,29 +38,46 @@ MPI_SCALARS     := { int;
                      unsigned long long int;
                      float;
                      double;
-                     long double }
+                     long double; 
+                     std::complex<double>;
```
Comment (jppelteret) [inline cmake/config/template-arguments.in]: These belong in some expansion list other than `DIFFERENTIABLE_REAL_SCALARS`.

BTW. Thanks for all of the work that you're doing on this!

**Comment by bangerth** [comment]
Just for reference, this of course increases the library. I get this:
```
BEFORE:
-rwxr-xr-x 1 bangerth fac 1348499024 Nov  4 08:44 libdeal_II.g.so.9.0.0-pre
-rwxr-xr-x 1 bangerth fac  150601848 Nov  4 08:43 libdeal_II.so.9.0.0-pre

AFTER:
-rwxr-xr-x 1 bangerth fac 1556985296 Nov  3 19:50 libdeal_II.g.so.9.0.0-pre
-rwxr-xr-x 1 bangerth fac  180397352 Nov  3 19:50 libdeal_II.so.9.0.0-pre
```

I think we really need to get those new vector types going! :-(

File: cmake/config/template-arguments.in (hunk 1 above)
Comment (bangerth) [reply cmake/config/template-arguments.in]: I don't think I understand whether you want me to change anything here?

Comment (jppelteret) [reply cmake/config/template-arguments.in]: I don't know how old the branch that you're working off of is, but I believe that @luca-heltai and I made [some alterations here a while ago](https://github.com/dealii/dealii/commit/f8b5289cf867e9493dfb04ee156d4d30577bd6b0) so I think that these changes are redundant or conflicting. So I think that these changes should be reverted. Here's the current state of these lines on `master`: 
```
// differentiable scalar types
DIFFERENTIABLE_REAL_SCALARS := {
		    @DEAL_II_EXPAND_TRILINOS_SACADO_TYPES@;
		    } 

// all possible scalar types
ALL_SCALAR_TYPES := {
    double; float ; std::complex<double>; std::complex<float>;
    @DEAL_II_EXPAND_TRILINOS_SACADO_TYPES@;
}
```

Comment (jppelteret) [reply cmake/config/template-arguments.in]: Ah, drats. Sorry, I missed some code folding in github. Please ignore!

**Comment by drwells** [comment]
Thats not so bad for adding more scalar types.

**Pull request merged**